src/agent/main.cpp

```diff
@@ -184,6 +193,10 @@ static int realmain(int argc, char *argv[])
             printRadioVersion = true;
             break;
 
+        case OTBR_OPT_AUTO_ATTACH:
+            enableAutoAttach = (optarg == nullptr) || atoi(optarg);
```

Comment:
What happens if `optarg` is not a valid int? 

I guess we need to catch the error by using `strtol` instead, and notify user of the invalid argument. 

Thoughts?

Reply:
I guess we need to avoid exception according to https://google.github.io/styleguide/cppguide.html#Exceptions

Thoughts? @bukepo

Reply:
Agreed not to use exceptions. Done with `strtol`.